src/discovery/lxd.go

```diff
@@ -0,0 +1,189 @@
+/**
+ * lxd.go - LXD API discovery implementation
+ *
+ */
+
+package discovery
+
+import (
+	"../config"
+	"../core"
+	"../logging"
+	"../utils"
+	"fmt"
+	"time"
+
+	"github.com/lxc/lxd"
+)
+
+const (
+	lxdRetryWaitDuration = 2 * time.Second
+	lxdTimeout           = 5 * time.Second
```

Comment:
Keeping in mind possible remote containers support, this could be configurable too